requests/utils.py

```diff
@@ -825,3 +825,19 @@ def rewind_body(prepared_request):
                                         "body for redirect.")
     else:
         raise UnrewindableBodyError("Unable to rewind request body for redirect.")
+
+def is_ascii(data):
```

Comment:
We could also rework this as:

``` python
    if isinstance(data, str):
        coding_func = getattr(data, 'encode')
    elif isinstance(data, bytes):
        coding_func = getattr(data, 'decode')
    else:
        return False

    # Try to encode/decode data into ascii
    try:
        coding_func('ascii')
        return True
    except (UnicodeError):
        return False
```

Either way it's kind of an ugly function but pulls that out of the method and hopefully can provide value elsewhere later.